arch/x86_64/fentry.S

```diff
@@ -1,8 +1,55 @@
 /* argument passing: %rdi, %rsi, %rdx, %rcx, %r8, %r9 */
 /* return value: %rax */
 /* callee saved: %rbx, %rbp, %rsp, %r12-r15 */
-/* stack frame (with -pg): return addr = 8(%rbp), prev fp = 0(%rbp) */
-/* stack frame (with -fentry): return addr = (%rsp), prev fp = 8(%rsp) */
+/* stack frame (with -pg -mfentry): parent addr = 56(%rsp), child addr = 48(%rsp) */
```

Comment:
Understood !
At the very beginning of `__fentry__()`,

parent addr (an address that has the return address of `hello()`): 8(%rsp)
child addr (the return address of `__fentry__()`): %rsp

Because the "%rbp" is not set yet.
Right ?